app/Jobs/ImportRockTheVotePosts.php

```diff
@@ -0,0 +1,459 @@
+    }
+
+    /*
+     * Translate "Opt-in to Partner SMS/robocall" from Rock the Vote CSV to a Northstar sms_status
+     *
+     * @param array $sms_status
+     * @return string
+    */
+    private function transformSmsStatus($sms_status)
+    {
+        if ($sms_status === 'Yes') {
+            return 'active';
+        }
+
+        // @TODO: do we want this to be 'pending' or some other status? we talked about this recently referring to something else
+        return 'stop';
+    }
+
+    /**
+     * For a given record and referral code values, first check if we have a northstar ID, then grab the user using that.
+     * Otherwise, see if we can find the user with the given email (if it exists), if not check if we can find them with the given phone number (if it exists).
+     * If all else fails, create the user .
+     *
+     * @TODO - If we have a northstar id in the referral code, then we probably don't need to make a call to northstar for the full user object.
+     *
+     * @return array
+     */
+    private function getOrCreateUser($record, $values)
+    {
+        $user = null;
+
+        $userFieldsToLookFor = [
+            'id' => isset($values['northstar_id']) && !empty($values['northstar_id']) ? $values['northstar_id'] : null,
+            'email' => isset($record['Email address']) && !empty($record['Email address']) ? $record['Email address'] : null,
+            'mobile' => isset($record['phone']) && !empty($record['phone']) ? $record['phone'] : null,
+        ];
+
+        foreach ($userFieldsToLookFor as $field => $value)
+        {
+            if ($value) {
+                info('getting user with the '.$field.' field', [$field => $value]);
+                $user = gateway('northstar')->asClient()->getUser($field, $value);
+            }
+
+            if ($user) {
+                break;
+            }
+        }
+
+        if (is_null($user)) {
+            $userData = [
+                'email' => $record['Email address'],
+                'mobile' => $record['Phone'],
+                'first_name' => $record['First name'],
+                'last_name' => $record['Last name'],
+                'addr_street1' => $record['Home address'],
+                'addr_street2' => $record['Home unit'],
+                'addr_city' => $record['Home city'],
+                'addr_state' => $record['Home state'],
+                'addr_zip' => $record['Home zip code'],
+                'source' => env('NORTHSTAR_CLIENT_ID'),
+            ];
+
+            if ($record['Phone']) {
+                $userData['sms_status'] = $this->transformSmsStatus($record['Opt-in to Partner SMS/robocall']);
+            }
+
+            $user = gateway('northstar')->asClient()->createUser($userData);
+
+            info('created user', ['user' => $user->id]);
+            $this->stats['countUserAccountsCreated']++;
+        }
+
+        return $user;
+    }
+
+    private function getCSVRecords($filepath)
```

Comment:
question! I noticed that this is also private in the [ImportTurboVotePosts Job](https://github.com/DoSomething/chompy/blob/master/app/Jobs/ImportTurboVotePosts.php#L387) as well and think I will need to use this exact same function again when I write the Facebook Share Import Job. Is there a reason this has to be private or can we make this a helper?

Reply:
@chloealee great question! I was looking into this but a lot of the functions had to be further customized, but I don't think that this one does! I'll make sure they match and pull it out!

Reply:
Okay so @chloealee since we're modifying instance variables in this one I decided to add this and `initStats()` as a trait called `ImportToRogue`. I don't _love_ traits because I think it's confusing to follow when you call `$this->{whatever}` and then the function is not in that file, but in this case hopefully it will help DRY things up and make Chompy overall easier to work with. We can always revert back! I also did not update the turbovote file because I'm not able to test that currently and we don't have any tests written .

Reply:
@katiecrane totally agree with you that sometimes traits can get a little confusing but love this solution here that you did! If it gets too hairy, we can change back like you said! This is super helpful though bc I definitely use those functions as well and we can now cut a lot of code in the jobs :)